File: LV4/4.4/cd/cd.cpp
#include <cstdio>
using namespace std;

int main() {
    int n, m;
    int A[1000010], B;

    while (scanf("%d %d", &n, &m), n || m) {
        int cnt = 0;
        int idx = 0;

        for (int i = 0; i < n; i++) {
            scanf("%d", &A[i]);
        }

        for (int i = 0; i < m; i++) {
            scanf("%d", &B);
            while (idx < n && B > A[idx]) {
                idx++;
            }
            if (idx < n && B == A[idx]) {
                cnt++;
            }
        }
        printf("%d\n", cnt);
    }
}

File: LV4/4.4/cd/cd.go
// +----------+
// | TOO SLOW |
// +----------+
package main

import (
	"fmt"
)

func main() {
	var n, m, B int
	A := make([]int, 1000010)

	for {
		fmt.Scanf("%d %d", &n, &m)
		if n == 0 && m == 0 {
			break
		}

		var idx, cnt int

		for i := 0; i < n; i++ {
			fmt.Scanf("%d", &A[i])
		}

		for i := 0; i < m; i++ {
			fmt.Scanf("%d", &B)
			for idx < n && B > A[idx] {
				idx++
			}
			if idx < n && B == A[idx] {
				cnt++
			}
		}
		fmt.Printf("%d\n", cnt)
	}
}
